### ISSUE TITLE
[Experiment][iOS] The same default browser homepage banner is displayed on all Treatment branches of the “Mobile default browser homepage banner copy test - iOS” experiment

### ISSUE THREAD

**Issue by alicevarro** [mozilla-mobile/firefox-ios#14751]
**[Additional Notes]:**
- Based on the [["Experiment: Set to default Homepage Banner with ECO body" ]](https://docs.google.com/document/d/1FMpXgBFKiu8rlqIN9YkjJMDQjNlqlSb74H4-Lu9CS4U/edit#)documentation, each treatment branch should have a different default browser homepage banner. However, I have verified all the branches and the same default browser homepage banner is displayed.
- However, we are not sure if the issue is reproducible because we are using the Preview collection or we are missing some steps.
- The link to the Experimenter page can be found [here](https://experimenter.services.mozilla.com/nimbus/mobile-default-browser-homepage-banner-copy-test-ios/summary).

**[Affected versions]:**
- Firefox Release 114.0 (31388 and 31383 builds downloaded from TestFlight).

**[Affected Platforms]:**
- iOS 16.2 - iPhone SE

**[Prerequisites]:**
- Have the Firefox build installed on the device.

**[Steps to reproduce]:**
1. Open the app from the prerequisites.
2. Tap the Kebab Menu > Settings > About Firefox.
3. Tap 5 times on the Firefox build.
4. Scroll down to the “Nimbus Experiments” section and tap on it.
5. Tap the “Edit” button and enable the “Use the Preview Collection” option.
6. Restart the Firefox app once or twice.
7. Navigate again to the “Nimbus Experiments” section.
8. Tap on the “Mobile default browser homepage banner copy test - iOS”  experiment.
9. If you are not enrolled in any of the branches, tap on a treatment branch.
8. Restart the app again.
9. Observe the elements displayed on the “Default browser” banner.

**[Expected result]:**
 - The specific Default homepage banner from the enrolled branch is displayed.

**[Actual result]:**
- The Control’s Default homepage banner is displayed even if you are enrolled in one of the Treatment branches.

**[Notes]:**
- Attaching a screen recording of the issue [here](https://drive.google.com/file/d/1KTCDICBZL3ZO2qgEMtd5S8994mC65Rgl/view?usp=sharing)

┆Issue is synchronized with this [Jira Task](https://mozilla-hub.atlassian.net/browse/FXIOS-6592)

**Comment by alicevarro** [comment]
I was also able to reproduce this issue using the "Mobile Default Browser CTA Copy test - iOS" experiment using the latest Firefox Beta 9000 (31606) and Firefox Release 115 (31602) builds downloaded from TestFlight. The Control’s Default homepage banner CTA is displayed even if you are enrolled in the Treatment branch.

I've noticed that by using a [clone of this experiment](https://stage.experimenter.nonprod.dataops.mozgcp.net/nimbus/mobile-default-browser-cta-copy-stage/summary) on the Stage server in a Firefox build set to Stage (on the latest main branch) we are no longer reproducing this issue.

**Comment by adudenamedruby** [comment]
THe reason this is happening, is that I'm not seeing the `surface` parameter in the experiment, so this is technically not a valid message.

<img width="767" alt="Screenshot 2023-06-15 at 11 13 24 AM" src="https://github.com/mozilla-mobile/firefox-ios/assets/11182210/c6a4e307-2479-4093-8884-fb4fd8d65dc0">

If you take a look at the `MessageData` [object](https://github.com/mozilla-mobile/firefox-ios/blob/b5afe6d21b2842defc54078e540c09d0e369f836/nimbus-features/messagingFeature.yaml#L139), you'll notice it expects a surface, whose default value is `unknown`, meaning that this message has nowhere to go. Presumably, you want this on the `new-tab-card` surface. Please update the experiment, re-test, and lemme know if this ticket can be closed. :)

**Comment by alicevarro** [comment]
Thank you for the explanations! 
I've updated the Stage experiment and it works as expected. Also the [Mobile Default Browser CTA Copy test - iOS](https://experimenter.services.mozilla.com/nimbus/mobile-default-browser-cta-copy-test-ios/summary) and [Mobile default browser homepage banner copy test - iOS](https://experimenter.services.mozilla.com/nimbus/mobile-default-browser-homepage-banner-copy-test-ios/summary) experiments from Prod were updated by Courtney and the correct homepage banners are displayed now.
This ticket can be closed now.
